docs/content/Examples-Tutorials-Recipes/Accelerating-Non-Additive-Measures.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Accelerating Non-Additive Measures
+permalink: /recipes/non-additivity
+category: Examples & Tutorials
+subCategory: Query acceleration
+menuOrder: 1
+---
+
+## Use case
+
+We want to run queries against
+[pre-aggregations](https://cube.dev/docs/caching#pre-aggregations) only to ensure our
+application's superior performance. Usually, accelerating a query is as simple as
+including its measures and dimensions to the pre-aggregation
+[definition](https://cube.dev/docs/schema/reference/pre-aggregations#parameters-measures).
+
+[Non-additive](https://cube.dev/docs/caching/pre-aggregations/getting-started#ensuring-pre-aggregations-are-targeted-by-queries-non-additivity)
+measures (e.g., average values or distinct counts) are a special case.
+Pre-aggregations with such measures are less likely to be
+[selected](https://cube.dev/docs/caching/pre-aggregations/getting-started#ensuring-pre-aggregations-are-targeted-by-queries-selecting-the-pre-aggregation)
+to accelerate a query. However, there are a few ways to work around that.
+
+## Data schema
+
+Let's explore the `Users` cube that contains various measures describing users' age:
+the count of unique age values (`distinctAges`), the average age (`avgAge`), and the
+90th [percentile](https://cube.dev/docs/recipes/percentiles) of age (`p90Age`).
+
+```javascript
+    distinctAges: {
+      sql: `age`,
+      type: `countDistinct`,
+    },
+
+    avgAge: {
+      sql: `age`,
+      type: `avg`,
+    },
+
+    p90Age: {
+      sql: `PERCENTILE_CONT(0.9) WITHIN GROUP (ORDER BY age)`,
+      type: `number`,
+    },
+```
+
+All of these measures are non-additive. Practically speaking, it means that the following pre-aggregation would only accelerate a query that fully matches its
+definition:
+
+```javascript
+    main: {
+      measures: [
+        CUBE.distinctAges,
+        CUBE.avgAge,
+        CUBE.p90Age
+      ],
+      dimensions: [
+        CUBE.gender
+      ]
+    },
+```
+
+So, this query will match the pre-aggregation above and, thus, will be accelerated:
+
+```javascript
+{
+  "measures": [
+    "Users.distinctAges",
+    "Users.avgAge",
+    "Users.p90Age"
+  ],
+  "dimensions": [
+    "Users.gender"
+  ]
+}
+```
+
+Meanwhile, this query won't match the same pre-aggregation because it contains non-addtive measures and omits the `gender` dimension. Thus, it won't be accelerated:
```

Comment:
```suggestion
Meanwhile, the query below won't match the same pre-aggregation because it contains non-additive measures and omits the `gender` dimension. It won't be accelerated:
```